src/common/paddle.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  type Environment,
+  Paddle,
+  type Subscription,
+} from '@paddle/paddle-node-sdk';
+import { logger } from '../logger';
+
+export const paddleInstance = new Paddle(process.env.PADDLE_API_KEY, {
+  environment: process.env.PADDLE_ENVIRONMENT as Environment,
+});
+
+type CancelSubscriptionProps = {
+  subscriptionId: string;
+};
+export const cancelSubscription = ({
+  subscriptionId,
+}: CancelSubscriptionProps): Promise<Subscription> | undefined => {
+  logger.info(
+    {
+      type: 'paddle',
+      subscriptionId,
+    },
+    'Subscription cancelled user deletion',
+  );
+  try {
+    return paddleInstance.subscriptions.cancel(subscriptionId, {
```

Comment:
```suggestion
    return await paddleInstance.subscriptions.cancel(subscriptionId, {
```

try catch will not catch error from promise that is not awaited